ale_linters/vala/vala_lint.vim

```diff
@@ -0,0 +1,64 @@
+" Author: Atsuya Takagi <asoftonight@gmail.com>
+" Description: A linter for Vala using Vala-Lint.
+
+call ale#Set('vala_vala_lint_enable_config', 0)
+call ale#Set('vala_vala_lint_config_filename', 'vala-lint.conf')
+
+function! ale_linters#vala#vala_lint#GetCommand(buffer) abort
+    let l:command = 'io.elementary.vala-lint '
+
+    if ale#Var(a:buffer, 'vala_vala_lint_enable_config')
```

Comment:
I don't think this vala_vala_lint_enable_config variable is needed. We can check if vala_vala_lint_config_filename is empty instead:

```
let l:config_filename = ale#Var(a:buffer, 'vala_vala_lint_config_filename')
let l:config_path = ale#path#FindNearestFile(a:buffer, l:config_filename)  " <- this is empty if l:config_filename is empty

if !empty(l:config_path)
    let l:command .= ' -c ' . l:config_path
endif

return l:command . ' %s'
```

Reply:
The reason for the enable flag was that I thought there might be a case where you want to keep your config filename value in your .vimrc but want to disable the config file search for some machines (i.e. on one machine, the config file is stored in network mounted filesystem, and searching through it can cause a delay).

But I think your suggestion makes sense as it makes things easier. Applied your suggested changes. Thanks!